### PR TITLE
[NO-TASK] added asset endpoint

### PR DESCRIPTION
**Pull request by Krysh90**

### CODE REVIEW

File: src/shared/models/asset/asset.controller.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { ApiBearerAuth, ApiCreatedResponse, ApiTags } from '@nestjs/swagger';
+import { GetJwt } from 'src/shared/auth/get-jwt.decorator';
+import { JwtPayload } from 'src/shared/auth/jwt-payload.interface';
+import { RoleGuard } from 'src/shared/auth/role.guard';
+import { WalletRole } from 'src/shared/auth/wallet-role.enum';
+import { AssetDto } from './asset.dto';
+import { Asset } from './asset.entity';
+import { AssetService } from './asset.service';
+
+@ApiTags('asset')
```
Comment (davidleomay) [inline src/shared/models/asset/asset.controller.ts:12]: `@ApiTags('Asset')`

File: src/shared/models/asset/asset.controller.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+export class AssetController {
+  constructor(private assetService: AssetService) {}
+
+  @Get()
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(WalletRole.USER))
+  @ApiCreatedResponse({ type: AssetDto, isArray: true })
+  async getAllAsset(@GetJwt() jwt?: JwtPayload): Promise<AssetDto[]> {
+    return this.assetService.getAllAssetsForBlockchain(jwt?.blockchain).then(this.toDto);
+  }
```
Comment (davidleomay) [inline src/shared/models/asset/asset.controller.ts:23]: If the AuthGuard is set, then jwt is never undefined.

File: src/shared/models/asset/asset.dto.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    description: 'Asset type of the asset',
+    enum: AssetType,
+  })
+  type: AssetType;
+
+  @ApiProperty({
+    description: 'Asset category of the asset',
+    enum: AssetCategory,
+  })
+  category: AssetCategory;
```
Comment (davidleomay) [inline src/shared/models/asset/asset.dto.ts:26]: Haha, good descriptions 😆

File: src/shared/models/asset/asset.controller.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  toDto(assets: Asset[]): AssetDto[] {
+    return assets.map((asset) => ({
+      name: asset.name,
+      displayName: asset.displayName,
+      type: asset.type,
+      category: asset.category,
+      blockchain: asset.blockchain,
+    }));
+  }
```
Comment (davidleomay) [inline src/shared/models/asset/asset.controller.ts:33]: In LOCK we started creating separate mapper classes, e.g. _src/subdomains/staking/application/mappers/withdrawal-output-dto.mapper.ts_

Comment (davidleomay) [reply src/shared/models/asset/asset.controller.ts:33]: I think we should add buyable/sellable columns to the entity, to be able to disable some assets.

File: src/shared/models/asset/asset.dto.ts (hunk 3 above)
Comment (Krysh90) [reply src/shared/models/asset/asset.dto.ts:26]: open for any suggestion :D

File: migration/1675928633077-asset-buyable-sellable.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        await queryRunner.query(`ALTER TABLE "sqldb-lock-api-dev"."dbo"."asset" ADD "buyable" bit NOT NULL CONSTRAINT "DF_02425ec147dfa126bfd6f41c7db" DEFAULT 0`);
+        await queryRunner.query(`ALTER TABLE "sqldb-lock-api-dev"."dbo"."asset" ADD "sellable" bit NOT NULL CONSTRAINT "DF_b5d5bb3aea25a3e735ec3c6a2fe" DEFAULT 0`);
+    }
+
+    async down(queryRunner) {
+        await queryRunner.query(`ALTER TABLE "sqldb-lock-api-dev"."dbo"."asset" DROP CONSTRAINT "DF_b5d5bb3aea25a3e735ec3c6a2fe"`);
+        await queryRunner.query(`ALTER TABLE "sqldb-lock-api-dev"."dbo"."asset" DROP COLUMN "sellable"`);
+        await queryRunner.query(`ALTER TABLE "sqldb-lock-api-dev"."dbo"."asset" DROP CONSTRAINT "DF_02425ec147dfa126bfd6f41c7db"`);
+        await queryRunner.query(`ALTER TABLE "sqldb-lock-api-dev"."dbo"."asset" DROP COLUMN "buyable"`);
+    }
```
Comment (davidleomay) [inline migration/1675928633077-asset-buyable-sellable.js:16]: Remove all database references (`"sqldb-lock-api-dev".`)

**Review by davidleomay — CHANGES_REQUESTED** [review]

**Review by davidleomay — APPROVED** [review]

**Pull request merged**